### PR TITLE
support config dns via network stanza

### PR DESCRIPTION
**Pull request by lisongmin**
Hi, I added the dns config support in this PR which defined in [job-specification/network#dns-parameters](https://www.nomadproject.io/docs/job-specification/network#dns-parameters).

This PR will resolve following scenario:

1. host is the dns provider, and fill `nameserver 127.0.0.1` in `/etc/resolv.conf`
2. running job with `containerd-driver`, app in task can't resolve the dns.

this PR requires building with nomad 1.0 or above, i don't know if it can run with nomad 0.10 or not. 
I just comfirm that it can running on nomad 1.0.2.

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
@lisongmin Looks like the `DNS support` was added in version `0.12.4` https://github.com/hashicorp/nomad/pull/7661
I think it's okay to update to nomad `1.0.2`

Can you check why the `circleci` tests are failing? Also, this might needs a rebase.

File: tests/006-test-dns.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+job_name=dns
+
+test_dns_nomad_job() {
+    pushd ~/go/src/github.com/Roblox/nomad-driver-containerd/example
+
+    nomad_major_version=$(nomad --version|sed -n 's/Nomad v\([0-9]\+\).*/\1/p')
+    if [ "$nomad_major_version" -le 0 ];then
+        # only support dns on nomad 1.0 and abover.
```
Comment (shishir-a412ed) [inline tests/006-test-dns.sh:10]: nitpick: `above`

File: tests/run_tests.sh [hunk 2]
```diff
@@ -72,12 +76,12 @@ setup() {
 
 	set +e
 	sudo pkill --signal SIGKILL -P $(ps faux | grep 'daily' | awk '{print $2}')
-	set -e
 
 	# Remove docker daemon and containerd.
 	sudo systemctl stop docker
 	sudo systemctl stop containerd
 	sudo apt-get purge -y docker-ce docker-ce-cli containerd.io
+	set -e
```
Comment (shishir-a412ed) [inline tests/run_tests.sh:84]: Why was `set -e` moved down?

File: tests/run_tests.sh [hunk 3]
```diff
@@ -2,11 +2,15 @@
 
 set -eo pipefail
 
-export NOMAD_VERSION=0.11.2
+export NOMAD_VERSION=1.0.2
 export CONTAINERD_VERSION=1.3.4
 export PATH=$PATH:/usr/local/go/bin
 export PATH=$PATH:/usr/local/bin
-export GOPATH=/home/circleci/go
+if [ -e /home/circleci ]; then
```
Comment (shishir-a412ed) [inline tests/run_tests.sh:9]: Is there a case where `/home/circleci` does not exist and we have to explicitly set `GOPATH=$HOME/go`?

File: example/dns.nomad [hunk 4]
```diff
@@ -0,0 +1,27 @@
+
+job "dns" {
+  datacenters = ["dc1"]
+
+  group "dns-group" {
+
+    network {
+        dns {
+            servers = ["127.0.0.1", "127.0.0.2"]
```
Comment (shishir-a412ed) [inline example/dns.nomad:9]: Can we also update the test to have `options` and `searches`?

```
network {
  dns {
    servers = ["127.0.0.1", "127.0.0.2"]
    searches = ["internal.corp"]
    options = ["ndots:2"]
  }
}
```

File: tests/006-test-dns.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+job_name=dns
+
+test_dns_nomad_job() {
+    pushd ~/go/src/github.com/Roblox/nomad-driver-containerd/example
+
+    nomad_major_version=$(nomad --version|sed -n 's/Nomad v\([0-9]\+\).*/\1/p')
+    if [ "$nomad_major_version" -le 0 ];then
```
Comment (shishir-a412ed) [inline tests/006-test-dns.sh:9]: If we are moving to nomad version `1.0.2` in `run_tests.sh`. Do we still need this check?

**Comment by shishir-a412ed** [comment]
@lisongmin There was some intermittent issue with `circleci` I have re-run the tests and now they are passing!

File: tests/run_tests.sh (hunk 2 above)
Comment (lisongmin) [reply tests/run_tests.sh:84]: How do you running tests locally? 

My steps is as following:

1. startup vagrant
2. run `./tests/run_tests.sh` to initial the test environment
3. modify and run test case until test pass

I need to change run_tests.sh so i can initial my test environment. 

Is there a better test step there?

File: tests/run_tests.sh (hunk 3 above)
Comment (lisongmin) [reply tests/run_tests.sh:9]: same as above

File: tests/006-test-dns.sh (hunk 5 above)
Comment (lisongmin) [reply tests/006-test-dns.sh]: removed.

File: example/dns.nomad (hunk 4 above)
Comment (lisongmin) [reply example/dns.nomad:9]: added.

File: tests/006-test-dns.sh (hunk 1 above)
Comment (lisongmin) [reply tests/006-test-dns.sh]: the block is removed.

File: tests/run_tests.sh (hunk 2 above)
Comment (shishir-a412ed) [reply tests/run_tests.sh:84]: aah I see your concern here 🙂
Initially, when I designed the tests, I wrote them to run only as part of a CI/CD system like `circleci`

See https://github.com/Roblox/nomad-driver-containerd/blob/master/tests/run_tests.sh#L17-L19
and 
https://github.com/Roblox/nomad-driver-containerd/blob/master/tests/run_tests.sh#L56

I will open a separate PR, so these tests can be run locally as well, inside the vagrant VM.

Comment (shishir-a412ed) [reply tests/run_tests.sh:84]: We can put `set -e` back to its original place for now.

File: tests/run_tests.sh (hunk 3 above)
Comment (shishir-a412ed) [reply tests/run_tests.sh:9]: This is okay. We can leave it as part of this PR.

**Review by shishir-a412ed — APPROVED** [review]
LGTM!

@lisongmin Will merge once [`this`](https://github.com/Roblox/nomad-driver-containerd/pull/65#discussion_r565533899) is addressed.

File: tests/run_tests.sh (hunk 2 above)
Comment (lisongmin) [reply tests/run_tests.sh]: done

**Pull request merged**